src/main/java/data/Recipe.java

```diff
@@ -0,0 +1,87 @@
+import java.util.LinkedList;
+import java.util.List;
+
+public class Recipe {
+
+  protected String name, description;
+  protected List<Step> steps;
+  private List<SpinOff> spinOffs;
+
+  public Recipe() {}
+  
+  public Recipe(String recipeName, String recipeDescription, List<Step> recipeSteps) {
+    this.name = recipeName;
+    this.description = recipeDescription;
+    this.steps = recipeSteps;
+    this.spinOffs = new LinkedList<>();
+  }
+
+  public void appendStep(Step newStep) {
+    steps.add(newStep);
+  }
+
+  public void addStep(int position, Step newStep) {
+    if (position < 0) {
+      return;
+    } else if (position >= steps.size()) {
+      appendStep(newStep);
+    } else {
+      steps.add(position, newStep);
+    }
+  }
+
+  public void setStep(int position, Step newStep) {
+    if (position < 0 || position > steps.size() - 1) {
+      return;
+    }
+    steps.set(position, newStep);
+  }
+
+  public void removeStep(int position) {
+    if (position < 0 || position > steps.size() - 1) {
+      return;
+    }
+    steps.remove(position);
+  }
+
+  protected void addSpinOff(SpinOff spinOff) {
+    spinOffs.add(spinOff);
```

Comment:
I thought I remembered discussing a feature where clicking on a recipe could show spin-offs made from that recipe. I think this would be an interesting way to show users how their recipes generate creativity, and would also contribute to a more social feel to Shef.

Reply:
Hmm, okay. I don't see it written down anywhere, but maybe I somehow missed it when I was writing everything down. We can talk about it at a next meeting (w Ryan :D)!